txcache/wrappedTransaction.go

```diff
@@ -1,72 +1,59 @@
 package txcache
 
 import (
-	"bytes"
+	"sync/atomic"
 
 	"github.com/multiversx/mx-chain-core-go/data"
 )
 
-const processFeeFactor = float64(0.8) // 80%
+// bunchOfTransactions is a slice of WrappedTransaction pointers
+type bunchOfTransactions []*WrappedTransaction
 
 // WrappedTransaction contains a transaction, its hash and extra information
 type WrappedTransaction struct {
-	Tx                   data.TransactionHandler
-	TxHash               []byte
-	SenderShardID        uint32
-	ReceiverShardID      uint32
-	Size                 int64
-	TxFeeScoreNormalized uint64
+	Tx              data.TransactionHandler
+	TxHash          []byte
+	SenderShardID   uint32
+	ReceiverShardID uint32
+	Size            int64
+
+	PricePerUnit atomic.Uint64
+	HashFnv32    atomic.Uint32
 }
 
-func (wrappedTx *WrappedTransaction) sameAs(another *WrappedTransaction) bool {
-	return bytes.Equal(wrappedTx.TxHash, another.TxHash)
-}
-
-// estimateTxGas returns an approximation for the necessary computation units (gas units)
-func estimateTxGas(tx *WrappedTransaction) uint64 {
-	gasLimit := tx.Tx.GetGasLimit()
-	return gasLimit
-}
-
-// estimateTxFeeScore returns a normalized approximation for the cost of a transaction
-func estimateTxFeeScore(tx *WrappedTransaction, txGasHandler TxGasHandler, txFeeHelper feeHelper) uint64 {
-	moveGas, processGas := txGasHandler.SplitTxGasInCategories(tx.Tx)
-
-	normalizedMoveGas := moveGas >> txFeeHelper.gasLimitShift()
-	normalizedProcessGas := processGas >> txFeeHelper.gasLimitShift()
-
-	normalizedGasPriceMove := txGasHandler.GasPriceForMove(tx.Tx) >> txFeeHelper.gasPriceShift()
-	normalizedGasPriceProcess := normalizeGasPriceProcessing(tx, txGasHandler, txFeeHelper)
-
-	normalizedFeeMove := normalizedMoveGas * normalizedGasPriceMove
-	normalizedFeeProcess := normalizedProcessGas * normalizedGasPriceProcess
+// precomputeFields computes (and caches) the (average) price per gas unit.
+func (wrappedTx *WrappedTransaction) precomputeFields(txGasHandler TxGasHandler) {
+	fee := txGasHandler.ComputeTxFee(wrappedTx.Tx).Uint64()
 
-	adjustmentFactor := computeProcessingGasPriceAdjustment(tx, txGasHandler, txFeeHelper)
-
-	tx.TxFeeScoreNormalized = normalizedFeeMove + normalizedFeeProcess*adjustmentFactor
-
-	return tx.TxFeeScoreNormalized
-}
+	gasLimit := wrappedTx.Tx.GetGasLimit()
+	if gasLimit == 0 {
+		return
```

Comment:
wouldn't it be better to set ppu to 0 in this case ?

Reply:
actually it is 0 by default.

Reply:
Indeed, zero by default.